### ISSUE TITLE
IconButton disabled state styling

### ISSUE THREAD

**Issue by pROFESOR11** [GeekyAnts/NativeBase#4625]
### Description

Can't apply styling to icons in IconButton when the state is `isDisabled`

### CodeSandbox/Snack link

https://snack.expo.dev/@profesor/nativebase-iconbutton-disabled-styling

### Steps to reproduce

Hi,
I'd like to decrease the opacity of IconButton when it is disabled. I was expecting styling in `_disabled` prop would be passed to the component specified in `icon` prop. Here is the code:

```
    <IconButton
          isDisabled
          onPress={handlePress}
          icon={
            // don't have disabled state here
            <Icon as={Ionicons} name="checkbox" size={10} color="green.700" />
          }
          _pressed={{
            bg: 'green.100',
          }}
          _icon={{
            // don't have _disabled option here
            _light: {
            },
          }}
          _disabled={{
            // don't have _icon option here
            // these styles are not passed to icon
            opacity: 0.4,
            style: {
              opacity: 0.4,
            },
          }}
    />
```

I'd like to report if this is a missing feature, since it is a bit edge-case, I'd be more than happy to hear if there is a way to do that with the current structure.

### NativeBase Version

3.3.4

### Platform

- [ ] Android
- [ ] CRA
- [X] Expo
- [X] iOS
- [ ] Next

**Comment by surajahmed** [comment]
@pROFESOR11 Thanks for reporting! This issue has been fixed in version [3.3.6](https://github.com/GeekyAnts/NativeBase/releases/tag/v3.3.6).

**Comment by trymbill** [comment]
Looks like `isDisabled` only places the icon in a disabled looking state, it doesn't prevent the icon button being clicked. The `onPress` callback still goes through. If you also provide it with a truthy `disabled` prop, the icon button isn't clickable. It's a bit awkward to need to provide the `<IconButton />` component with both `isDisabled` and `disabled` at the same time.
